dashboard/apps/prototype/models.py

```diff
@@ -212,7 +227,61 @@ def money_spent(self, start_date, end_date, contractor_only=False,
             tasks = tasks.all()
         spending_per_task = [task.money_spent(start_date, end_date)
                              for task in tasks]
-        return sum(spending_per_task)
+        costs = self.cost_between(start_date, end_date)
+        return sum(spending_per_task) + costs
+
+    def cost_between(self, start_date, end_date):
+        def cost_of_cost(cost):
+            return cost.cost_between(start_date, end_date)
+
+        return sum(map(cost_of_cost, self.costs.filter(
+            Q(end_date__lte=end_date) | Q(end_date__isnull=True),
+            start_date__gte=start_date
+        ))) or Decimal('0')
+
+
+class Cost(models.Model):
+    project = models.ForeignKey('Project', related_name='costs')
+    start_date = models.DateField()
+    end_date = models.DateField(null=True, blank=True)
+    name = models.CharField(max_length=128, null=True)
+    note = models.TextField(null=True, blank=True)
+    cost = models.DecimalField(max_digits=10, decimal_places=2)
+    type = models.PositiveSmallIntegerField(
+        choices=COST_TYPES, default=COST_TYPES.ONE_OFF)
+
+    def cost_between(self, start_date, end_date):
+        start_date = max(start_date, self.start_date)
+        end_date = min(end_date, self.end_date or end_date)
+        if self.type == COST_TYPES.ONE_OFF and \
+           start_date <= self.start_date <= end_date:
+            return self.cost
+        if self.type == COST_TYPES.MONTHLY:
+            freq = MONTHLY
+        elif self.type == COST_TYPES.ANNUALLY:
+            freq = YEARLY
+        dates = dates_between(start_date, end_date, freq)
+        return len(dates) * self.cost
+
+
+class Budget(models.Model):
+    project = models.ForeignKey('Project', related_name='budgets')
+    start_date = models.DateField()
+    budget = models.DecimalField(max_digits=16, decimal_places=2)
+
```

Comment:
This is just in terms of our application and for our mvp it will just be used for graphing budget etc so end_date isn't needed